src/executors/collect_battery.h

```diff
@@ -0,0 +1,51 @@
+/*
+ * Created by xiaofei.lan@rokid.com on 2019/04/26.
+ */
+
+#ifndef YODA_SIXSIX_COLLECT_BATTERY_H
+#define YODA_SIXSIX_COLLECT_BATTERY_H
+
+#include "executor_def.h"
+
+#define CHARGER_PATH "/sys/class/power_supply/bq25890-charger"
+#define BATTERY_PATH "/sys/class/power_supply/battery"
+
+#define GET_CHARGER_FILE(path) CHARGER_PATH path
+#define GET_BATTERY_FILE(path) BATTERY_PATH path
+
+using namespace std;
+
+YODA_NS_BEGIN
+
+class CollectBattery : public IJobExecutor {
+public:
+  CollectBattery();
+
+  ~CollectBattery() override;
+
+  void execute() override;
+
+  int stop() override;
+
+protected:
+
+  virtual void doCollect(uv_work_t *req);
```

Comment:
The keyword `virtual` is actually redundant, no other class inherits this class. I forget to remove it in `CollectTop` and `CollectSmap`. :-)